stringutils/stringutils.go

```diff
@@ -0,0 +1,158 @@
+package stringutils
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	DefaultPrecision      = 2
+	DefaultBase           = 10
+	DefaultFormat         = 'f'
+	DefaultNilMapFormat   = "{}"
+	DefaultNilSliceFormat = "[]"
+	DefaultNilFormat      = "<nil>"
+)
+
+// Options - stringification options.
+type Options struct {
+	Base           int
+	Format         byte
+	Precision      int
+	NilFormat      string
+	NilMapFormat   string
+	NilSliceFormat string
+}
+
+func parseOptions(opts ...Options) Options {
+	options := Options{
+		Base:           DefaultBase,
+		Precision:      DefaultPrecision,
+		Format:         DefaultFormat,
+		NilFormat:      DefaultNilFormat,
+		NilMapFormat:   DefaultNilMapFormat,
+		NilSliceFormat: DefaultNilSliceFormat,
+	}
+
+	for _, opt := range opts {
+		if opt.Base != 0 {
+			options.Base = opt.Base
+		}
+		if opt.Precision != 0 {
+			options.Precision = opt.Precision
+		}
+		if opt.Format != uint8(0) {
+			options.Format = opt.Format
+		}
+		if opt.NilFormat != "" {
+			options.NilFormat = opt.NilFormat
+		}
+		if opt.NilMapFormat != "" {
+			options.NilMapFormat = opt.NilMapFormat
+		}
+		if opt.NilSliceFormat != "" {
+			options.NilSliceFormat = opt.NilSliceFormat
+		}
+	}
+
+	return options
+}
+
+// Stringify receives an arbitrary value and converts it into a string.
+// Stringify also accepts an options object with the following properties:
+// 	Options.NilFormat: the string format for nil values, defaults to "<nil>".
+// 	Options.NilMapFormat: the string format for nil map objects, defaults to "{}".
+// 	Options.NilSliceFormat: the string format for nil slice objects, defaults to "[]".
+//	Options.Base: a number between 2-36 ad the base when converting ints and uints to strings, defaults to Base 10.
+// 	Options.Precision: number of digits to include when converting floats and complex numbers to strings, defaults to 2.
+// 	Options.Format: the number notation format, using the stlib ftoa functionalities, defaults to 'f':
+// 		'b' (-ddddp±ddd, a binary exponent),
+// 		'e' (-d.dddde±dd, a decimal exponent),
+// 		'E' (-d.ddddE±dd, a decimal exponent),
+// 		'f' (-ddd.dddd, no exponent),
+// 		'g' ('e' for large exponents, 'f' otherwise),
+// 		'G' ('E' for large exponents, 'f' otherwise),
+// 		'x' (-0xd.ddddp±ddd, a hexadecimal fraction and binary exponent), or
+// 		'X' (-0Xd.ddddP±ddd, a hexadecimal fraction and binary exponent).
+func Stringify(value any, opts ...Options) string {
+	options := parseOptions(opts...)
+
+	// first use a type switch, because its more performant
+	switch assigned := value.(type) {
+	case string:
+		return assigned
+	case bool:
+		strconv.FormatBool(assigned)
+	case nil:
+		return options.NilFormat
+	case []byte:
+		return string(assigned)
+	case int:
```

Comment:
can group the cases that share the same return together and use `fallthrough`, to reduce the amount of repetition since the int conversions are the same and the uint conversions are the same.